### ISSUE TITLE
Support python3

### ISSUE THREAD

**Issue by ngoldbaum** [andrewshadura/git-crecord#1]
It seems that this only works under python2. Unfortunately I use python3 as my daily driver, it would be nice to work under both python2 and python3.

**Comment by andrewshadura** [comment]
Sure, I plan this as the next step when I have staging and unstaging done.
This comes from Mercurial targeting Python 2, and I decided to skip porting for now so that I can concentrate on more important things.

**Comment by pklapperich** [comment]
Well, Python2 end of life is less than 2 years away and if pypi and distros start to remove python2 support at that point, things could get difficult if we don't start porting now. So I did.

I have a [python3 branch in my fork](/pklapperich/git-crecord/tree/python3) where I started porting. Most of the work was related to python3 changing strings from bytes to utf-8. It works, but has not been well tested (on either python2 or 3) and at this time I don't recommend merging it with master.

It adds a dependency on the `future` module (`pip install future`). This dependency can be removed if python2 support is dropped.

It installs fine with `python3 setup.py install`. I've only tested on a Linux system that uses UTF-8 encoding in the terminal; it might not work on other systems. I've run it a bit with coverage (from `pip3 install coverage`) , but there's still a lot of code paths I haven't run:

```
Name                            Stmts   Miss  Cover
---------------------------------------------------
git-crecord                         9      0   100%
git_crecord/__init__.py             0      0   100%
git_crecord/chunk_selector.py     601    371    38%
git_crecord/crecord_core.py       142     59    58%
git_crecord/crpatch.py            409    229    44%
git_crecord/encoding.py           224    165    26%
git_crecord/gitrepo.py             51     11    78%
git_crecord/main.py               145     71    51%
git_crecord/util.py                77     40    48%
---------------------------------------------------
TOTAL                            1658    946    43%
```

It's possible my branch has broken things in either python2 or python3 or both. We probably want some automated test scripts that get full code coverage.
